KenticoOnboardingApplication/test/KenticoOnboardingApplication.Api.Tests/Controllers/ListControllerIntegrationTest.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+using KenticoOnboardingApplication.Api.Controllers;
+using KenticoOnboardingApplication.Contracts.Helpers;
+using KenticoOnboardingApplication.Contracts.Models;
+using KenticoOnboardingApplication.Contracts.Repositories;
+using KenticoOnboardingApplication.Tests.Base;
+using KenticoOnboardingApplication.Tests.Base.Factories;
+using NSubstitute;
+using NUnit.Framework;
+using Unity;
+
+namespace KenticoOnboardingApplication.Api.Tests.Controllers
+{
+    [TestFixture]
+    public class ListControllerIntegrationTest
+    {
+        [Test]
+        public async Task PutItemAsync_ShouldCallRepositoryOnce()
+        {
+            var container = CreateContainer();
+            var controller = ResolveController(container);
+            var (repository, timeManager) = ResolveRepositoryAndTimeManager(container);
+            var (originItem, expectedItem, itemToPut) = CreateOriginExpectedAndItemToPut();
+            ArrangeRepositoryAndTimeManager(repository, timeManager, originItem, expectedItem);
+
+            var result = await controller.PutItemAsync(itemToPut.Id, itemToPut);
+            var executedResult = await result.ExecuteAsync(CancellationToken.None);
+            executedResult.TryGetContentValue(out Item resultItem);
+
+            await repository.Received(1).GetItemAsync(originItem.Id);
+            await repository.DidNotReceive().GetItemAsync(Arg.Is<Guid>(id => id != originItem.Id));
+            await repository.Received(1).UpdateItemAsync(Arg.Is<Item>(item => item.Id == originItem.Id));
+            await repository.DidNotReceive().UpdateItemAsync(Arg.Is<Item>(item => item.Id != originItem.Id));
+            await repository.DidNotReceive().GetAllItemsAsync();
+            await repository.DidNotReceive().DeleteItemAsync(Arg.Any<Guid>());
+            await repository.DidNotReceive().AddItemAsync(Arg.Any<Item>());
```

Comment:
This block of `repository` assertions form kind of a, hm, ..., I would say, … block 😆 – let's move them to a private method as well.